db.js

```diff
@@ -3355,167 +3431,234 @@ module.exports.CreateDB = function (parent, func) {
+                obj.performingBackup = false;
+                backupStatus = 0x0;
+            });
+            output.on('end', function () { });
+            output.on('error', function (err) {
+                if ((backupStatus & BACKUPFAIL_ZIPCREATE) == 0) {
+                    console.log('Output error: ' + err);
+                    if (func) { func('Output error: ' + err); };
+                    backupStatus |= BACKUPFAIL_ZIPCREATE;
+                    archive.abort();
+                };
+            });
+            archive.on('warning', function (err) {
+                //if files added to the archiver object aren't reachable anymore (e.g. sqlite-journal files)
+                //an ENOENT warning is given, but the archiver module has no option to/does not skip/resume
+                //so the backup needs te be aborted as it otherwise leaves an incomplete zip and never 'ends'
+                if ((backupStatus & BACKUPFAIL_ZIPCREATE) == 0) {
+                    console.log('Zip warning: ' + err); 
+                    if (func) { func('Zip warning: ' + err); };
+                    backupStatus |= BACKUPFAIL_ZIPCREATE;
+                    archive.abort();
+                };
+            });
+            archive.on('error', function (err) {
+                if ((backupStatus & BACKUPFAIL_ZIPCREATE) == 0) {
+                    console.log('Zip error: ' + err);
+                    if (func) { func('Zip error: ' + err); };
+                    backupStatus |= BACKUPFAIL_ZIPCREATE;
+                    archive.abort();
+                }
                 });
-            }
-        } catch (ex) { console.log(ex); }
-        return 0;
-    }
+            archive.pipe(output);
+
+            let globIgnoreFiles;
+            //slice in case exclusion gets pushed
+            globIgnoreFiles = parent.config.settings.autobackup.backupignorefilesglob.slice();
+            if (parent.config.settings.sqlite3) { globIgnoreFiles.push (datapathFoldername + '/' + databaseName + '.sqlite*'); }; //skip sqlite database file, and temp files with ext -journal, -wal & -shm
+            //archiver.glob doesn't seem to use the third param, archivesubdir. Bug?
+            //workaround: go up a dir and add data dir explicitly to keep the zip tidy
+            archive.glob((datapathFoldername + '/**'), {
+                cwd: datapathParentPath,
+                ignore: globIgnoreFiles,
+                skip: parent.config.settings.autobackup.backupskipfoldersglob
+            });
+
+            if (parent.config.settings.autobackup.backupwebfolders) {
+                if (parent.webViewsOverridePath) { archive.directory(parent.webViewsOverridePath, 'meshcentral-views');; }
+                if (parent.webPublicOverridePath) { archive.directory(parent.webPublicOverridePath, 'meshcentral-public');; }
+                if (parent.webEmailsOverridePath) { archive.directory(parent.webEmailsOverridePath, 'meshcentral-emails');; }
+            };
```

Comment:
double `;;` at the ending above, my OCD is annoying, please fix

Reply:
Can't have that! Removed.